coderd/database/dbmem/dbmem.go

```diff
@@ -3296,34 +3296,51 @@ func (q *FakeQuerier) GetFilteredInboxNotificationsByUserID(_ context.Context, a
 	defer q.mutex.RUnlock()
 
 	notifications := make([]database.InboxNotification, 0)
-	for _, notification := range q.inboxNotifications {
+	for idx := len(q.inboxNotifications) - 1; idx >= 0; idx-- {
+		notification := q.inboxNotifications[idx]
```

Comment:
nit: could simplify with https://pkg.go.dev/slices#Backward

```suggestion
	for _, notification := range slices.Backward(q.inboxNotifications) {
```

Reply:
Thanks, was definitely not aware about this one. 
If i'm not wrong - I just checked and seems it has been introduced with go `v1.23` - our go.mod is set to `v1.22.9`. 🥲

Reply:
Perhaps drop a FIXME comment for future yourself :)